test/test_mixeddofhandler.jl

```diff
@@ -643,6 +643,26 @@ function test_vtk_export()
     rm(filename*".vtu") # clean up
 end
 
+function test_celliterator_on_true_subdomain_smoketest()
+    grid = generate_grid(Hexahedron, (3,3,3))
+
+    dh = DofHandler(grid)
+    sdh = SubDofHandler(dh, [1,2,3])
+    add!(sdh, :u, Lagrange{RefHexahedron,1}()^3)
+    close!(dh)
+
+    # The folloging statements just check that the iterator
+    # does not crash at least. Regression for #966
+    for cell in CellIterator(sdh)
+    end
+
+    for cell in CellIterator(dh, [1,2,3])
+    end
+
+    for cell in CellIterator(dh)
+    end
```

Comment:
```suggestion
        if cellid(cell) <= 3
            @test length(celldofs(cell)) == getnbasefunctions(ip)
        else
            @test length(celldofs(cell)) == 0
        end
    end
```

Reply:
Do we really want to enforce this behavior?

Reply:
I don't think we should document it, but it is good to be aware if a later change changes this behavior.